eng/scripts/Copy-ApiVersion-Functions.ps1

```diff
@@ -1,19 +1,27 @@
-function Get-NewTagSection($apiVersion, $resourceProvider, $apiVersionStatus, $specFiles) {
-    return @"
-### Tag: package-$apiVersion
+function Get-NewTagSection($apiVersion, $resourceProvider, $apiVersionStatus, $specFiles, $resourceProviderInnerFolder = $null) {
 
-These settings apply only when ``--tag=package-$apiVersion`` is specified on the command line.
+    $tagVersion = $apiVersion -match '(?<date>\d{4}-\d{2})-\d{2}' 
+    $tagVersion = $Matches['date']
+    $baseDir = if ($resourceProviderInnerFolder) 
+    { "$resourceProvider/$resourceProviderInnerFolder/$apiVersionStatus/$apiVersion" } 
+    else { "$resourceProvider/$apiVersionStatus/$apiVersion" }
+    
+    $content = @"
+### Tag: package-$(If($apiVersionStatus -eq "preview"){"preview-"})$tagVersion
 
-```````yaml `$(tag) == 'package-$apiVersion'
-input-file:
-$(
-foreach ($specFile in $specFiles) {
-    "  - $resourceProvider/$apiVersionStatus/$apiVersion/$specFile"
-}
-)
-```````
+These settings apply only when ``--tag=package-$(If($apiVersionStatus -eq "preview"){"preview-"})$tagVersion`` is specified on the command line.
 
+```````yaml `$(tag) == 'package-$(If($apiVersionStatus -eq "preview"){"preview-"})$tagVersion'`
+input-file:
 "@
+
+    foreach ($specFile in $specFiles) {
+        $content += "`n  - $baseDir/$specFile"
+    }
+
+    $content += "`n```````n"
```

Comment:
This works, but you might consider something for better readability e.g.,

```pwsh
$content += "`n" + '```' + "`n"
```

Single quotes are literals, so you don't need to escape the backtick, which means you don't have to carefully count them.

Reply:
Thanks for the tip!